projects/trustednode/index.js

```diff
@@ -0,0 +1,91 @@
+const sdk = require('@defillama/sdk')
+const { getChainTransform } = require('../helper/portedTokens')
+const { unwrapUniswapLPs } = require('../helper/unwrapLPs')
+const bn = require("bn.js")
+
+const bscVaults = [
+    {
+        name: "TNODE VAULT",
+        chain: "bsc",
+        vaultAddr: "0x98386F210af731ECbeE7cbbA12C47A8E65bC8856",
+        contractAddr: "0x7f12a37b6921ffac11fab16338b3ae67ee0c462b",
+        isLp: false,
+    },
+    {
+        name: "BSC LIQUIDITY VAULT TNODE/BUSD",
+        chain: "bsc",
+        vaultAddr: "0x44dC7FE8e51076De1B9f863138107148b441853C",
+        contractAddr: "0x562C0c707984D40b98cCba889C6847DE274E5d57",
+        isLp: true,
+    },
+]
+
+const fantomVaults = [
+    {
+        name: "FTM LIQUIDITY VAULT TNODE/USDC",
+        chain: "fantom",
+        vaultAddr: "0xe056aba40572f64d98a8c8e717c34e96056c4aad",
+        contractAddr: "0x9206444A1820c508FbA5bF815713451Ee540B3c8",
+        isLp: true,
+    },
+]
+
+async function vaultTvl(timesatamp, block, chainBlocks, vaults) {
+    const balances = {}
+
+    for (const vault of vaults) {
+        const totalSupply = (await sdk.api.abi.call({
+            abi: 'erc20:totalSupply',
```

Comment:
can we count deposits to the vault instead of total supply?

Reply:
Hi @waynebruce0x I'm TL at Trusted Node, this function `totalSupply` on our vaults contracts has the same name as the ERC20 method, but it actually returns the token amounts deposited by users. It's required for differentiating between assets in the vault owned by users, and assets in the vault owned by the vault (rewards that have yet to be distributed). The distinction wouldn't be necessary in the case that the staking token isn't the same as the rewards token, but for one of our vaults they are the same.

To clear up confusion I propose that we would specify our own ABI for this method, instead of reusing the ERC20 one, even if the end effect is the same), or add a comment here specifying that this totalSupply call is returning something different. What do you think?